src/sctools/barcode.py

```diff
@@ -112,3 +113,66 @@ def from_iterable_bytes(cls, iterable, barcode_length):
         """construct an ObservedBarcodeSet from an iterable of bytes barcodes"""
         tbe = TwoBit(barcode_length)
         return cls(Counter(tbe.encode(b) for b in iterable), barcode_length=barcode_length)
+
+
+class ErrorsToCorrectBarcodesMap:
+
+    def __init__(self, errors_to_barcodes):
+        """use a hash map of erroneous_barcode -> true barcode to correct errors in cell barcodes
+
+        :param dict errors_to_barcodes:
+        """
+        if not isinstance(errors_to_barcodes, Mapping):
+            raise TypeError('errors_to_barcodes must be a mapping of erroneous barcodes to correct '
+                            'barcodes, not %a' % type(errors_to_barcodes))
+        self._map = errors_to_barcodes
+
+    def correct_barcode(self, barcode):
```

Comment:
Maybe rename this to `get_correct_barcode` to clarify that this just returns the correct barcode